docs/changelog/0037-upgrade-to-elasticsearch7.md

```diff
@@ -0,0 +1,336 @@
+  ```
+  # to start user-pillow (replace with UserPillow if that's what you use)
+    cchq <env> service pillowtop start --only user-pillow
+    cchq <env> service pillowtop stop --only user-pillow
+    
+  ```
+3. At this point, your ES2 hosts should not be getting any HTTP requests.
+  - You can confirm this by doing a tcpdump on Elasticsearch port on any of your Elasticsearch 2 hosts
+  ```
+  tcpdump -A -s 0 'tcp port 9200 and (((ip[2:2] - ((ip[0]&0xf)<<2)) - ((tcp[12]&0xf0)>>2)) != 0)'_
+  ```
+4. Reindex by running below command for each of the index in`['forms', 'cases', 'users', 'domains', 'apps', 'groups', 'sms', 'report_cases', 'report_xforms', 'case_search']`
+  ```
+  ./manage.py reindex_from_es2_to_es7 <index_name> <es2_host>
+  ```
+  - These can be run concurrently in different tmux/screen sessions
+  - Note that these management commands are to be run from the release that was created in Setup phase, which is pointing to Elasticsearch 7
+5. At this point, all of the Elasticsearch 2 data should be in Elasticsearch 7.
+   - You can confirm this by running the `reindex_from_es2_to_es7` management command with `print_index_size` option which displays number of documents in both the ES clusters. 
+7. Run below command to update Elasticsearch settings on ES7 indices for usage
+  ```
+  ./manage.py initialize_es_indices --set-for-usage
+  ```
+8. Update HQ config to point to Elasticsearch 7.
+  - Update below settings in you env's public.yml in addition to other setting changes made in upgrade-branch in the Setup phase
+  ```
+  ELASTICSEARCH_MAJOR_VERSION: 7
+  ```
+  - Update config
+  ```
+  cchq <env> update-config --branch=<upgrade-branch>
+  ```
+    - Merge the upgrade-branch into master  
+9. Make sure that HQ is able to connect to Elasticsearch 7
+  ```
+  cchq <env> django-manage check_services
+  ```
+10. End downtime
+  ```
+  cchq <env> downtime end
+  ```
+That's it, you are running HQ on Elasticsearch 7!
+
+
+
+### Reindex and Switchover separately
+
+If your form/case index sizes are large, you can follow this method to first reindex without downtime and then switchover with downtime.
+
+#### Reindex
+
+Reindex phase involves reindexing data from Elasticsearch 2 cluster into new Elasticsearch 7 cluster. Form and Case indices can be indexed in background while the HQ application continues to be live. This reindex can be performed based on the `inserted_at` date field upto a day before switchover.  Using this parameter multiple reindex processes can be run concurrently for different date ranges enabling you to reindex very large indices. The smaller offset data inserted newly during and after the reindex phase can be reindexed  during the last phase of switchover with downtime. This enables minimising the overall downtime required to upgrade drastically. 
+
+Use below notes to reindex all of the data upto a day or two before switchover and proceed to the switchover section after that.
```

Comment:
"upto" should be two words "up to" throughout this file (it is used in multiple places).